### ISSUE TITLE
Configurable option: Disabling root folder & classifier creation [feature request]

### ISSUE THREAD

**Issue by jelenadreskai** [MauroDataMapper/mdm-core#245]
As an administrator, I want all users (including container administrators) to be restricted when creating folders and classifiers, so the data model organization structure is controlled by the main MDM provider/administrators.

Acceptance criteria:

- Non-administrator (content administrators/readers/editors/authors etc] should not be able to create a root folder. If content administrator, he/she should be allowed to create folders within their folders but not outside of their autorized zone 
- Non-administrator (content administrators/readers/editors/authors etc] should not be able to create classifier

This could a configurable option, so certain MDM instances may want to allow this option, whereas others would like to restrict it.

**Comment by jelenadreskai** [comment]
@olliefreeman Here we go. This is a feature request. I cannot set any tags, you will have to tag it as a feature.

**Comment by olliefreeman** [comment]
i envision this as an ApiProperty which would be then configurable via the admin properties page.
At the moment any authenticated users can create a root folder/classifier (container), this option would be checked in the interceptors and return a forbidden response, which is what a non-logged in user gets, if the option is enabled.

If the option is enabled then only users in groups which have been assigned the application_administor (or site_administrator) role would be able to create root containers

The security permissions are such that if a user is in a group with the container_administrator role assigned to a container then they can "mess" with that container. This option would NOT stop that, the groups would have to be configured such that users were in groups that were at max editor roles on the containers, this would allow them to create/import models inside the containers but not change the container in anyway.

To be able to move a container from one location to another the user has to have container_Admin rights on the folder being moved and the folder being moved into. There is however a caveat to that where the user only has to have container_admin rights to move a container up to the root location. 
This would be something we could prevent so that the user has to have container_admin rights on the folder being moved and be in a usergroup with the application_administrator role (which by proxy would also automatically give them container_admin rights on all containers anyway)

It would be disabled by default as our normal behaviour should continue.

**Comment by jelenadreskai** [comment]
@olliefreeman, we have @GBishop-PHSA already working on this? We can contribute the changes once they are done.

**Comment by olliefreeman** [comment]
@jelenadreskai @GBishop-PHSA no-one is working on this at the moment, i was about to start to but i;ll assign it to gail

**Comment by olliefreeman** [comment]
just to be clear the default should definitely be that non-admin users can create folders and classifiers at the root.

**Comment by olliefreeman** [comment]
@GBishop-PHSA 
wrt question https://github.com/MauroDataMapper/mdm-core/issues/306#issuecomment-1130318959

the property should be `maurodatamapper.security.restrict.root.containers` as this ticket should cover both classifiers and folders...therfore ALL containters

**Comment by jelenadreskai** [comment]
Hi @olliefreeman. Are you sure you want these 2 under the same configuration? We want both, but I am not sure about other users. They may want to allow non-admins to create classifiers, but not folders. 

I would suggest having 2 separate variables:
security.restrict.root.folder
security.restrict.root.classifier (we could remove root, if there is no plan to have classifiers as a tree)

Thanks,
Jelena
